### PR TITLE
Add Laravel support

### PR DESCRIPTION
**Pull request by FunkyOz**
Adding laravel support and remove Symfony
Update docs to reflect new chages.

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (dunglas) [inline .github/workflows/ci.yml]: PHPUnit is installed by default with Laravel, so you can remove this line.

File: Dockerfile
Comment (dunglas) [inline Dockerfile]: By curiosity, why do you replace Caddy with NGINX? Caddy supports Laravel very well (one of Caddy's maintainers is a Laravel developer), supports much more features than NGINX (HTTP/3, auto-HTTPS...), and allows a much simpler Docker setup.

File: docker-compose.override.yml
Comment (dunglas) [inline docker-compose.override.yml]: It looks like all features you use are supported in version 3.8, so you could keep 3.4 to support a larger set of Docker Compose versions.

File: docker-compose.override.yml
Comment (dunglas) [inline docker-compose.override.yml]: This works for Laravel too :)

File: docker-compose.override.yml
Comment (FunkyOz) [reply docker-compose.override.yml]: Sure you are right, I didn't think about

File: Dockerfile
Comment (FunkyOz) [reply Dockerfile]: Because I don't know it verry well so I prefer manage nginx I've worked with. So I don't close this review until I studied and re add Caddy. Thank's for think about! :)

File: .github/workflows/ci.yml
Comment (FunkyOz) [reply .github/workflows/ci.yml]: Removed! thk

File: Dockerfile
Comment (FunkyOz) [reply Dockerfile]: Hi Kévin, I re add Caddy as web server. It's very useful and simple, thanks again!

**Comment by FunkyOz** [comment]
Hi Kévin, can you approve the review if everything is ok? Thanks!

File: Dockerfile
Comment (dunglas) [inline Dockerfile]: Nitpicking: I would remove this line to prevent future useless merge conflicts if you want to import change we make upstream.

File: Dockerfile
Comment (dunglas) [inline Dockerfile:49]: Do you know what takes so long? This can maybe be optimized?

File: Dockerfile
Comment (dunglas) [inline Dockerfile]: Psaume here for merge conflicts.

File: docker-compose.yml
Comment (dunglas) [inline docker-compose.yml]: You should use the same indentation style again to prevent merge conflicts.

File: docker-compose.yml
Comment (dunglas) [inline docker-compose.yml]: I'm not sure of what these changes are doing but maybe should we make them upstream too.

File: docker/caddy/Caddyfile
Comment (dunglas) [inline docker/caddy/Caddyfile]: It's perfectly fine to remove Mercure, but for the record it is supported by Laravel: https://github.com/mvanduijker/laravel-mercure-broadcaster

File: docker/php/docker-entrypoint.sh
Comment (dunglas) [inline docker/php/docker-entrypoint.sh:22]: Indentation issues?

File: docs/build.md
Comment (dunglas) [inline docs/build.md:41]: You can restore these lines now.

**Review by dunglas — APPROVED** [review]

File: Dockerfile
Comment (FunkyOz) [reply Dockerfile]: No nitpicking, in fact thanks again :)
Do you refer to the line removed?

File: Dockerfile
Comment (FunkyOz) [reply Dockerfile:49]: Yes i thinks about it but laravel run some scripts after installation of framework and dependencies. In symfony/skeleton installation does not have scripts after so it take much less time.

I try to optimaze it removing scripts runs in tmp dir and move this logic after copying in root dir (inside docker-entrypoint.sh).

So in the worst case, when a new project starts, it takes around 90s so I decide to start in 120s for security. Healthcheck starts after container was started and start-period option describe when to start counting interval/timeout logic so in the worst case (new project) the healthcheck is resolved correctly before 120s.
I decided this value considering that maybe network should have problems.

https://stackoverflow.com/questions/53289950/clarification-on-docker-composes-start-period-parameter

File: docker-compose.yml
Comment (FunkyOz) [reply docker-compose.yml]: You're rigth, I accidentally override .editorconfig using that of laravel. I'll change it thk!

File: docker-compose.yml
Comment (FunkyOz) [reply docker-compose.yml]: Option depends_on is useful for container startup control. https://docs.docker.com/compose/startup-order/

In my opinion web server container should be start after setting of php container. Does not change much for usage cause until php is not ready we cannot start developing, even when web server is not setted.
In this way we declare this dependencies and if we need to do more action in php before starting web sever we can do it and web server will wait for it.

File: docker/php/docker-entrypoint.sh
Comment (FunkyOz) [reply docker/php/docker-entrypoint.sh:22]: Very strange because in my IDE it's correctly indentig.

Screenshot 2023-08-15 at 21 09 17

**Pull request merged**
